pandas/core/groupby/groupby.py

```diff
@@ -1363,19 +1366,28 @@ def _numba_agg_general(
 
         data = self._obj_with_exclusions
         df = data if data.ndim == 2 else data.to_frame()
-        starts, ends, sorted_index, sorted_data = self._numba_prep(df)
+
+        # starts, ends, sorted_index, sorted_data = self._numba_prep(df)
+
+        sorted_df = df.take(self.grouper._sort_idx, axis=self.axis)
+        sorted_ids = self.grouper._sorted_ids
+        _, _, ngroups = self.grouper.group_info
+        starts, ends = lib.generate_slices(sorted_ids, ngroups)
         aggregator = executor.generate_shared_aggregator(
-            func, **get_jit_arguments(engine_kwargs)
+            func, dtype_mapping, **get_jit_arguments(engine_kwargs)
         )
-        result = aggregator(sorted_data, starts, ends, 0, *aggregator_args)
+        result = sorted_df._mgr.apply(
+            aggregator, start=starts, end=ends, **aggregator_kwargs
+        )
```

Comment:
This is changing `*arggregator_args -> **aggregator_kwargs`, but then within `aggregator` it is being used as `*aggregator_kwargs`. This is only used internally right? I'm just wondering if we can make this less fragile somehow (changing of order kwargs might produce a bug, right?), but I'm not seeing a way.

Reply:
Yeah, the current method is really sketchy but it should be OK, since UDFs take another path.
(only args/kwargs that go through here are stuff like ``ddof`` for std/var).

The reason it's like this is since ``BlockManager.apply`` only takes kwargs.
Is it fine to change that?